test MOdule/testOCSP.pl

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env perl
+
+use warnings;
+use strict;
+
+use Socket;
+use Net::SSLeay qw(die_now die_if_ssl_error);
+
+use IO::Socket::SSL qw(debug0);
+
+use Time::gmtime;
+use Time::ParseDate;
+use HTTP::Tiny;
+use Data::Dumper;
+Net::SSLeay::load_error_strings();
+Net::SSLeay::SSLeay_add_ssl_algorithms();
+Net::SSLeay::randomize();
+
+my ($dest_serv, $port) = ("www.migros.ch","443");    # Read command line
+$port = getservbyname( $port, 'tcp' ) unless $port =~ /^\d+$/;
+	my $dest_ip = gethostbyname($dest_serv);
+	my $dest_serv_params = sockaddr_in( $port, $dest_ip );
+
+	socket( S, &AF_INET, &SOCK_STREAM, 0 ) or die "socket: $!";
+	connect( S, $dest_serv_params ) or die "connect: $!";
+	select(S);
+	$| = 1;
+	select(STDOUT);                   # Eliminate STDIO buffering
+
+	# The network connection is now open, lets fire up SSL
+
+	my $ctx = Net::SSLeay::CTX_new() or die_now("Failed to create SSL_CTX $!");
+	Net::SSLeay::CTX_set_options( $ctx, &Net::SSLeay::OP_ALL )
+	  and die_if_ssl_error("ssl ctx set options");
+	my $ssl = Net::SSLeay::new($ctx) or die_now("Failed to create SSL $!");
+	Net::SSLeay::set_fd( $ssl, fileno(S) );    # Must use fileno
+	my $res = Net::SSLeay::connect($ssl) and die_if_ssl_error("ssl connect");
+
+# get id(s) for given certs, like from get_peer_certificate
+# or get_peer_cert_chain. This will croak if
+# - one tries to make an OCSP_CERTID for a self-signed certificate
+# - the issuer of the certificate cannot be found in the SSL objects
+#   store, nor in the current certificate chain
+my $cert = Net::SSLeay::get_peer_certificate($ssl);
+
+my $id = eval { Net::SSLeay::OCSP_cert2ids($ssl,$cert) };
+die "failed to make OCSP_CERTID: $@" if $@;
+
+# create OCSP_REQUEST from id(s)
+# Multiple can be put into the same request, if the same OCSP responder
+# is responsible for them.
+my $req = Net::SSLeay::OCSP_ids2req($id);
+ 
+# determine URI of OCSP responder
+my $uri = Net::SSLeay::P_X509_get_ocsp_uri($cert);
+# Send stringified OCSP_REQUEST with POST to $uri.
+# We can ignore certificate verification for https, because the OCSP
+# response itself is signed.
+my $ua = HTTP::Tiny->new(verify_SSL => 0);
+$res = $ua->request( 'POST',$uri, {
+    headers => { 'Content-type' => 'application/ocsp-request' },
```

Comment:
Hello,

Question for the external content on website (Facebook, google plus, twitter, etc…).
As I can see, every web site (bank, etc…) we have link to social network… It is pertinent to check this?

I try to check with this code… I parse the web page content and check if there is the word facebook.com <http://facebook.com/>, plus.google.com <http://plus.google.com/>, etc..
it’s right or you mean it’s not the right way to do this?

thanks

And to check if there is a https redirect on the http request I catch every redirect and test the final if the location in the HTTP Header is https://. I don’t know if I have to do this or to use an other trick. Do you know Whatweb? I discover this and I think it’s a good workaround to do this check...

sub check_content{
	my ($host) = @_;
	my $check = {};
	$logger->info(" - Info: Cheking content");
	# Create a user agent object
	my $agent = LWP::UserAgent->new(env_proxy => 1,keep_alive => 1, timeout => 30);
	$agent->max_redirect(5);
	$agent->agent("Mozilla/5.0 (X11; Ubuntu; Linux i686; rv:39.0) Gecko/20100101 Firefox/39.0");
	my $url = "http://$host/";

	$logger->info(" - Info: Cheking HTTP Header");
	$logger->info(" - Info: Check if HTTPs redirect on HTTP request");

	my $response = $agent->get($url);
	my $request = $response->request();
	my @redirects = $response->redirects();
	$count = @redirects;
	$check->{host} =$host;
	my $i =0 ;
	print $count;
	foreach my $res (@redirects) {
		my $req = $res->request();
		print $res->header("Server");print "\n";
		print $res->status_line();
		if($res->header("Server")){$check->{srv_type}=$res->header("Server");print $res->header("Server");print "\n";};

		if ($i == $count-1 && $res->header("Location") =~ m/https/){

			$logger->info(" - Info: Check if Strict-Transport-Security is implemented");
			if($res->header("Strict-Transport-Security")){

				$check->{https_redirect} = "HTTPs redirect OK";
				$check->{hsts} = "HSTS ok";
			}else{
				$check->{https_redirect} = "HTTPs redirect OK";
				$check->{hsts} = "No HSTS";
			}
		} elsif($i == $count-1 && $res->header("Location") =~ m/http/){
			$check->{https_redirect} = "No HTTPs redirect";
			$check->{hsts} = "No HSTS";
		}
		$i++;
	}

Behar

> Le 17 août 2015 à 07:04, Jeanneret Cedric <notifications@github.com> a écrit :
> 
> In test MOdule/testOCSP.pl <https://github.com/EthACKdotOrg/swiss-ssl-lawn/pull/2#discussion_r37158623>:
> 
> > +my $id = eval { Net::SSLeay::OCSP_cert2ids($ssl,$cert) };
> > +die "failed to make OCSP_CERTID: $@" if $@;
> > +
> > +# create OCSP_REQUEST from id(s)
> > +# Multiple can be put into the same request, if the same OCSP responder
> > +# is responsible for them.
> > +my $req = Net::SSLeay::OCSP_ids2req($id);
> > +
> > +# determine URI of OCSP responder
> > +my $uri = Net::SSLeay::P_X509_get_ocsp_uri($cert);
> > +# Send stringified OCSP_REQUEST with POST to $uri.
> > +# We can ignore certificate verification for https, because the OCSP
> > +# response itself is signed.
> > +my $ua = HTTP::Tiny->new(verify_SSL => 0);
> > +$res = $ua->request( 'POST',$uri, {
> > +    headers => { 'Content-type' => 'application/ocsp-request' },
> Attention: certains serveurs OCSP (comme celui de StartSSL) fonctionnent avec SNI, il faudrait donc passer le bon header pour t'assurer que tout marche bien.
> Tu peux tester le serveur www.ethack.org <http://www.ethack.org/>, qui utilise StartSSL, pour avoir leur OCSP.
> 
> La commande OpenSSL pour faire la query chez eux:
> 
> openssl ocsp -issuer ../startssl/sub.class2.server.ca.pem -header "HOST" "ocsp.startssl.com" -cert certificate.pem -resp_text -url http://ocsp.startssl.com/sub/class2/server/ca
> Je te laisse faire en sorte de porter ça dans ton code ;).
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/EthACKdotOrg/swiss-ssl-lawn/pull/2/files#r37158623>.
>